File: DDD5_UI/src/views/cms/villager/villager/villagerService.js

import axios from 'axios';

/**
 * 村民信息
 */
export class VillagerService {
    // 分页查询列表
    static findVillagers (queryParams) {
        return axios({
            url: '/cms/villager/Villager/findVillagers',
            method: 'POST',
            data: queryParams
        });
    }
    //新增保存村民信息
    static saveVillager (villager) {
        const query = {}
        return axios({
            url: '/cms/villager/Villager/saveVillager',
            method: 'POST',
            params: query,
            data: villager
        });
    }

    //修改保存村民信息
    static updateVillager (villager) {
        const query = {}
        return axios({
            url: '/cms/villager/Villager/updateVillager',
            method: 'POST',
            params: query,
            data: villager
        });
    }
}

File: DDD5/ems/src/main/java/com/mt/cms/service/villager/VillagerService.java
package com.mt.cms.service.villager;

import com.mt.cms.entity.villager.Villager;
import com.mt.common.core.web.base.PageDTO;
import com.mt.common.core.web.base.PageResultDTO;

public interface VillagerService {
    /**
     * 根据分页参数查询村民信息集合
     *
     * @param pageDTO 分页条件
     */
    public PageResultDTO findVillagers(PageDTO pageDTO);

    /**
     * 新增村民信息
     *
     * @param villager 实体对象
     */
    public Villager saveVillager(Villager villager);

    /**
     * 更新村民信息
     *
     * @param villager 实体对象
     */
    public Villager updateVillager(Villager villager);

    /**
     * 根据ID删除村民信息
     *
     * @param villagerId ID
     */
    public void deleteVillager(Long villagerId);
}
